scripts/await-previous-builds.js

```diff
@@ -0,0 +1,73 @@
+// @ts-check
+// basic idea came from https://github.com/bellkev/circle-lock-test
+
+const fetch = require("node-fetch").default
+const chalk = require("chalk").default
+const { throttle } = require("lodash")
+
+/**
+ * Returns a promise that resolves once all previous circleCI builds on the current branch
+ * have finished (or failed for whatever reason).
+ * @param {{pollInterval?: number, logInterval?:number}?} props
+ */
+async function awaitPreviousBuilds({ pollInterval = 5000, logInterval = 30000 } = {}) {
+  const { CIRCLE_TOKEN, CIRCLE_BRANCH, CIRCLECI } = process.env
+
+  if (CIRCLECI !== "true") {
+    throw new Error("awaitPreviousBuilds only works on CI")
+  }
+
+  if (!CIRCLE_TOKEN) {
+    throw new Error(
+      "Please add CIRCLE_TOKEN in env. Must have read access for the project. See https://circleci.com/docs/api/#authentication (make sure it's scope: all)"
+    )
+  }
```

Comment:
Do we already expose this token? If not, will it be shared with forks? And, if so, what abilities does it grant and are we ok granting those others?

Also, seeing as this is specifically about doing releases, it seems to me that in reality this env variable would only need to be exposed for `master`, right?

Reply:
This will be shared with forks, but it's only granting read access to the circle API for this repo, so that's safe to do, I think. See https://circleci.com/docs/2.0/managing-api-tokens/#overview

But yeah this key is only needed on master. Not sure how to share secrets only with particular branches, but it might be possible. Do you think it's worth figuring out?

Reply:
Well, if it’s only read access then that seems fine to me. Perhaps good to just know in general, but that’s not specific to this PR. /cc @zephraph